server/src/test/java/org/elasticsearch/cluster/routing/AllocationIdIT.java

```diff
@@ -224,15 +232,17 @@ private void putFakeCorruptionMarker(IndexSettings indexSettings, ShardId shardI
     }
 
     private void checkNoValidShardCopy(String indexName, ShardId shardId) throws Exception {
-        final ClusterAllocationExplanation explanation =
-            client().admin().cluster().prepareAllocationExplain()
-                .setIndex(indexName).setShard(shardId.id()).setPrimary(true)
-                .get().getExplanation();
-
-        final ShardAllocationDecision shardAllocationDecision = explanation.getShardAllocationDecision();
-        assertThat(shardAllocationDecision.isDecisionTaken(), equalTo(true));
-        assertThat(shardAllocationDecision.getAllocateDecision().getAllocationDecision(),
-            equalTo(AllocationDecision.NO_VALID_SHARD_COPY));
+        assertBusy(() -> {
```

Comment:
was this a test bug before?

Reply:
Looks like it, although I think Zen2's faster elections exposed it.